sunpy/coordinates/utils.py

```diff
@@ -387,6 +387,103 @@ def get_rectangle_coordinates(bottom_left, *, top_right=None,
+
+    @classmethod
+    def from_corners(cls, bottom_left: SkyCoord, top_right: SkyCoord, as_tuple: bool = False):
+        """
+        Create a rectangle from the bottom-left and top-right coordinates.
+        If `as_tuple=True`, return a tuple of (bottom_left, top_right) instead of a Rectangle object.
+        """
+        top_right = top_right.transform_to(bottom_left)
+        if as_tuple:
+            return (bottom_left, top_right)
+        return cls(bottom_left, top_right)
+
+    @classmethod
+    def from_bottom_left(cls, bottom_left: SkyCoord, width: u.deg, height: u.deg, as_tuple: bool = False):
+        """
+        Create a rectangle from the bottom-left corner, width, and height.
+        If `as_tuple=True`, return a tuple of (bottom_left, top_right) instead of a Rectangle object.
+        """
+        # Validate width and height
+        if width < 0 * u.deg:
+            raise ValueError("The specified width cannot be negative.")
+        if width > 360 * u.deg:
+            raise ValueError("The specified width cannot be greater than 360 degrees.")
+        if height < 0 * u.deg:
+            raise ValueError("The specified height cannot be negative.")
+        if bottom_left.spherical.lat + height > 90 * u.deg:
+            raise ValueError("The specified height exceeds the maximum latitude.")
+
+        top_right = SkyCoord(
+            bottom_left.spherical.lon + width,
+            bottom_left.spherical.lat + height,
+            frame=bottom_left.frame
+        )
+
+        if as_tuple:
+            return (bottom_left, top_right)
+        return cls(bottom_left, top_right)
+
+    @classmethod
+    def from_center(cls, center: SkyCoord, width: u.deg, height: u.deg, as_tuple: bool = False):
+        """
+        Create a rectangle from the center coordinate, width, and height.
+        If `as_tuple=True`, return a tuple of (bottom_left, top_right) instead of a Rectangle object.
+
+        Parameters
+        ----------
+        center : `~astropy.coordinates.SkyCoord`
+            The center coordinate of the rectangle.
+        width : `~astropy.units.Quantity`
+            The width of the rectangle.
+        height : `~astropy.units.Quantity`
+            The height of the rectangle.
+        as_tuple : bool, optional
+            If True, return a tuple of coordinates. Otherwise, return a Rectangle object.
+        """
+        if width < 0 * u.deg:
+            raise ValueError("The specified width cannot be negative.")
+        if width > 360 * u.deg:
+            raise ValueError("The specified width cannot be greater than 360 degrees.")
+        if height < 0 * u.deg:
+            raise ValueError("The specified height cannot be negative.")
+        if center.spherical.lat + height / 2 > 90 * u.deg or center.spherical.lat - height / 2 < -90 * u.deg:
+            raise ValueError("The specified height exceeds the maximum latitude.")
+
+        bottom_left = SkyCoord(
+            center.spherical.lon - width / 2,
```

Comment:
Should we call these width and height, considering that we are defining the rectangle from the center? It's implied that the topmost point would be at half the distance from the center.